src/internal/js/mod.rs

```diff
@@ -307,10 +313,26 @@ impl JsString {
         }
     }
 
+    pub fn ucs2_value(self) -> Vec<u16> {
```

Comment:
I suspect (but didn't verify) it's because external strings are modeled as weak references that need special handling by the garbage collector.  Normal strings are simply collected when unreachable, external strings need to have their destructor invoked.